drivers/at86rf215/at86rf215_getset.c

```diff
@@ -29,17 +29,42 @@
 /* we can still go +3 dBm higher by increasing PA current */
 #define PAC_DBM_MIN                             (-31)     /* dBm */
 
-uint16_t at86rf215_get_addr_short(const at86rf215_t *dev)
+uint16_t at86rf215_get_addr_short(const at86rf215_t *dev, uint8_t filter)
 {
-    return at86rf215_reg_read16(dev, dev->BBC->RG_MACSHA0F0);
+    if (filter > 3){
+        return 0; 
+    }
+    return byteorder_ntohs((network_uint16_t)at86rf215_reg_read16(dev, dev->BBC->RG_MACSHA0F0 + (4*filter)));
+
 }
 
-void at86rf215_set_addr_short(at86rf215_t *dev, uint16_t addr)
+void at86rf215_set_addr_short(at86rf215_t *dev, uint8_t filter, uint16_t addr)
 {
-    dev->netdev.short_addr[0] = (uint8_t)(addr);
-    dev->netdev.short_addr[1] = (uint8_t)(addr >> 8);
+    if (filter == 0){  //dot not overwrite node address for filters other than 0 
+        dev->netdev.short_addr[0] = (uint8_t)(addr);
+        dev->netdev.short_addr[1] = (uint8_t)(addr >> 8);
+    }
+    if (filter > 3){
+        return; 
+    }
+    at86rf215_reg_write16(dev, dev->BBC->RG_MACSHA0F0 + (4*filter), byteorder_htons(addr).u16);
+}
 
-    at86rf215_reg_write16(dev, dev->BBC->RG_MACSHA0F0, addr);
+uint8_t at86rf215_get_framefilter_enabled(at86rf215_t *dev, uint8_t filter)
+{
+    return (at86rf215_reg_read(dev, dev->BBC->RG_AFC0) >> (AFC0_AFEN0_SHIFT + filter)) & 1;
+}
+
+void at86rf215_set_framefilter_enabled(at86rf215_t *dev, uint8_t filter, uint8_t state)
```

Comment:
since state is either `0` or `1` and not bitmap, just use `bool` to avoid confusion.

Might also make a cleaner API to just have `at86rf215_enable_framefilter(dev, filter)` & `at86rf215_disable_framefilter(dev, filter)`.